pkg/api/v1alpha1/cluster_types.go

```diff
@@ -200,6 +200,7 @@ type WorkerNodeGroupConfiguration struct {
 }
 
 func generateWorkerNodeGroupKey(c WorkerNodeGroupConfiguration) (key string) {
+	key = c.Name
 	if c.MachineGroupRef != nil {
 		key = c.MachineGroupRef.Kind + c.MachineGroupRef.Name
```

Comment:
Can we remove this? Now all groups should have a name
I feel like this method can just be `return c.Name`

Reply:
if we're using this key we need the name to be a part of it; otherwise, a wngc with the same machine ref and count but a different name would return the same key.